File: malipo-php/src/Exceptions/MalipoException.php
<?php

namespace Malipo\Exceptions;

use Exception;

class MalipoException extends Exception
{
    protected $statusCode;
    protected $errorCode;
    protected $details;

    public function __construct(string $message, int $statusCode = 0, ?string $errorCode = null, array $details = [])
    {
        parent::__construct($message);
        $this->statusCode = $statusCode;
        $this->errorCode = $errorCode;
        $this->details = $details;
    }

    public function getStatusCode(): int
    {
        return $this->statusCode;
    }

    public function getErrorCode(): ?string
    {
        return $this->errorCode;
    }

    public function getDetails(): array
    {
        return $this->details;
    }
}

File: malipo-java/src/main/java/com/malipo/exceptions/MalipoException.java
package com.malipo.exceptions;

import java.util.Map;

public class MalipoException extends RuntimeException {
    private final int statusCode;
    private final String errorCode;
    private final Map<String, Object> details;

    public MalipoException(String message, int statusCode, String errorCode, Map<String, Object> details) {
        super(message);
        this.statusCode = statusCode;
        this.errorCode = errorCode;
        this.details = details;
    }

    public int getStatusCode() { return statusCode; }
    public String getErrorCode() { return errorCode; }
    public Map<String, Object> getDetails() { return details; }

    @Override
    public String toString() {
        return "MalipoException{" +
                "message='" + getMessage() + '\'' +
                ", statusCode=" + statusCode +
                ", errorCode='" + errorCode + '\'' +
                ", details=" + details +
                '}';
    }
}
